File: app/src/main/java/com/maksk993/cryptoportfolio/presentation/view/activities/MainActivity.java
package com.maksk993.cryptoportfolio.presentation.view.activities;

import androidx.annotation.NonNull;
import androidx.appcompat.app.AppCompatActivity;
import androidx.fragment.app.Fragment;
import androidx.fragment.app.FragmentManager;
import androidx.fragment.app.FragmentTransaction;
import androidx.lifecycle.Lifecycle;
import androidx.lifecycle.LifecycleOwner;
import androidx.lifecycle.Observer;
import androidx.lifecycle.ViewModelProvider;

import android.os.Bundle;
import android.view.MenuItem;

import com.maksk993.cryptoportfolio.R;
import com.maksk993.cryptoportfolio.presentation.contract.MainContract;
import com.maksk993.cryptoportfolio.presentation.models.FindFragmentById;
import com.maksk993.cryptoportfolio.presentation.presenter.MainPresenter;
import com.google.android.material.bottomnavigation.BottomNavigationView;
import com.maksk993.cryptoportfolio.presentation.viewmodel.MainViewModel;

public class MainActivity extends AppCompatActivity {
    MainViewModel viewModel;
    FragmentManager fragmentManager = getSupportFragmentManager();
    BottomNavigationView bottomNavigationView;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        viewModel = new ViewModelProvider(this).get(MainViewModel.class);

        bottomNavigationView = findViewById(R.id.bottom_nav_menu);
        bottomNavigationView.setOnNavigationItemSelectedListener(new BottomNavigationView.OnNavigationItemSelectedListener() {
            @Override
            public boolean onNavigationItemSelected(@NonNull MenuItem item) {
                int id = item.getItemId();
                int fragmentId;

                if (id == R.id.nav_history) fragmentId = FindFragmentById.getInteger(FindFragmentById.HISTORY);
                else if (id == R.id.nav_portfolio) fragmentId = FindFragmentById.getInteger(FindFragmentById.PORTFOLIO);
                else fragmentId = FindFragmentById.getInteger(FindFragmentById.SETTINGS);

                viewModel.openFragment(fragmentId);
                return true;
            }
        });

        viewModel.shouldNewFragmentBeOpened().observe(this, new Observer<FindFragmentById>() {
            @Override
            public void onChanged(FindFragmentById findFragmentById) {
                Fragment fragment = FindFragmentById.getFragment(findFragmentById);
                FragmentTransaction transaction = fragmentManager.beginTransaction();
                transaction.replace(R.id.nav_host_fragment, fragment);
                transaction.commit();
            }
        });
    }
}

File: app/src/main/java/com/maksk993/cryptoportfolio/presentation/view/activities/MainActivity.kt
package com.maksk993.cryptoportfolio.presentation.view.activities

import android.os.Bundle
import androidx.activity.viewModels
import androidx.appcompat.app.AppCompatActivity
import androidx.appcompat.app.AppCompatDelegate
import com.google.android.material.bottomnavigation.BottomNavigationView
import com.maksk993.cryptoportfolio.R
import com.maksk993.cryptoportfolio.databinding.ActivityMainBinding
import com.maksk993.cryptoportfolio.presentation.models.FindFragmentById
import com.maksk993.cryptoportfolio.presentation.viewmodel.MainViewModel
import dagger.hilt.android.AndroidEntryPoint


@AndroidEntryPoint
class MainActivity : AppCompatActivity() {
    private val viewModel: MainViewModel by viewModels()
    private lateinit var binding : ActivityMainBinding

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = ActivityMainBinding.inflate(layoutInflater)
        setContentView(binding.root)

        viewModel.apply {
            startReceivingData()
            getAddedAssetsFromDb()
            getTransactionsFromDb()
            setSavedAppTheme()
        }

        initBottomNavMenu()
        initObservers()
    }

    private fun initObservers(){
        viewModel.nextFragment.observe(this) {
            val fragment = FindFragmentById.getFragment(it)
            val transaction = supportFragmentManager.beginTransaction()
            transaction.replace(R.id.nav_host_fragment, fragment)
            transaction.commit()
        }

        viewModel.nightMode.observe(this){
            var isNightMode = false
            if (it == true){
                AppCompatDelegate.setDefaultNightMode(AppCompatDelegate.MODE_NIGHT_YES)
                binding.btnChangeAcc.setImageResource(R.drawable.ic_arrow_down)
                isNightMode = true
            }
            else {
                AppCompatDelegate.setDefaultNightMode(AppCompatDelegate.MODE_NIGHT_NO)
                binding.btnChangeAcc.setImageResource(R.drawable.ic_arrow_down_night)
            }
            viewModel.setNightTheme(isNightMode)
        }
    }

    private fun initBottomNavMenu(){
        binding.bottomNavMenu.setOnNavigationItemSelectedListener(BottomNavigationView.OnNavigationItemSelectedListener
        {
            val id = it.itemId
            val fragmentId: FindFragmentById =
                when (id) {
                    R.id.nav_history -> FindFragmentById.HISTORY
                    R.id.nav_portfolio -> FindFragmentById.PORTFOLIO
                    else -> FindFragmentById.SETTINGS
                }
            viewModel.openFragment(fragmentId)
            true
        })
    }
}